ci-operator/config/maistra/istio/maistra-istio-maistra-3.0.yaml

```
@@ -62,13 +74,71 @@ tests:
       env:
       - name: MAISTRA_NAMESPACE
       - name: MAISTRA_SC_POD
-      from: src
+      from: src-test
       resources:
         requests:
           cpu: 100m
           memory: 100Mi
       timeout: 2h0m0s
     workflow: servicemesh-istio-e2e-aws
+- as: maistra-istio-unit-3-0
+  commands: |
+    make -e T=-v build racetest binaries-test \
+    BUILD_WITH_CONTAINER="0" \
+    GOBIN="/gobin" \
+    GOCACHE="/tmp/cache" \
+    GOMODCACHE="/tmp/cache" \
+    XDG_CACHE_HOME="/tmp/cache" \
+    ARTIFACTS="${ARTIFACT_DIR:-}" \
+    2>&1 | tee "${ARTIFACT_DIR}"/make-unit.log
+  container:
+    from: src-test
+- as: integ-pilot-3-0
+  cluster_claim:
+    architecture: amd64
+    cloud: aws
+    owner: openshift-ci
+    product: ocp
+    timeout: 1h0m0s
+    version: "4.10"
+  steps:
+    env:
+      MAISTRA_BUILDER_IMAGE: registry.ci.openshift.org/ci/maistra-builder:3.0
+      MAISTRA_NAMESPACE: maistra-e2e-test
+      MAISTRA_SC_POD: maistra-e2e-test-sc-pod
+    test:
+    - as: copy-src
+      cli: latest
+      commands: |
+        # SRC_PATH does end with /. : the content of the source directory is copied into dest directory
+        oc cp ./. "${MAISTRA_NAMESPACE}"/"${MAISTRA_SC_POD}":/work/
```

```suggestion
        # The following target directory only works in 3.0 job. 
        # For previous release, copy source to the remote directory /work
        oc cp ./. "${MAISTRA_NAMESPACE}"/"${MAISTRA_SC_POD}":/
```

If we are using upstream builder image, their workdir is root directory.
Matt voted a workaround by copying source to remote pod root directory.